pkg/controller/imagebuild/component/builddispatcher.go

```diff
@@ -104,7 +104,15 @@ func (c *BuildDispatcherComponent) Reconcile(ctx *core.Context) (ctrl.Result, er
 	}(configDir)
 
 	validateCredsSeg := txn.StartSegment("credentials-validate")
-	if err = credentials.Verify(ctx, configDir); err != nil {
+
+	insecureRegistries := make([]string, 0)
+	for reg, opts := range c.cfg.Registries {
+		if opts.Insecure || opts.HTTP {
+			insecureRegistries = append(insecureRegistries, reg)
+		}
+	}
```

Comment:
There's a part of me that wants to use `Config.Validate` to disallow registry entries with both insecure/http fields false so we wouldn't have to perform this check here, but I think your design is more flexible and still works. Nothing to change here, just thinking via comments.